content/outcomes/application-development/pair-programming.md

```diff
@@ -0,0 +1,128 @@
+
+### Go with the Flow
+Just go for it! Sit down, start pairing, and ride whatever driver/navigator pattern emerges. The pair should be disciplined about switching roles, and be cognisant of whether they are falling into an unbalanced driver/navigator pattern. 
+
+### Pomodoro 
+[Pomodoro](https://en.wikipedia.org/wiki/Pomodoro_Technique) is a dependable study technique where a timer is used to signal the switch between roles, taking breaks, or even switching pairs entirely. Typically, this is 25 minutes of work, followed by a 5-minute break.
+
+### TDD Ping-Pong 
+“Ping-Pong” pairing is a natural complement to [Test Driven Development (TDD)](https://en.wikipedia.org/wiki/Test-driven_development)**.** The classic Ping Pong technique first has the driver write a failing test for some desired functionality that is not yet implemented. The pair switches roles, having the new driver implement the minimum necessary code in order to pass the test. That driver then writes the next failing test and the cycle continues, with each person equally sharing test-writing and functionality-implementation duties.
+
+### Switch Roles on Research
+Switch roles any time the pair wants to research something further. This helps reduce the impulse to separate and research individually, which can sometimes evolve into an excuse to stop pairing entirely. 
+
+### Pairmate/One-keyboard
+A small item like a stuffed animal or tchotchke can be used to signal who is navigating. The navigator holds the item to signal that they are not to touch the keyboard until that item is passed off (physically or virtually when remote) to the other. Alternatively, the item can be removed,and the pair shares a single keyboard or toggles remote-keyboard control as needed.
+
+### Task List
+The pair decides on a set of tasks that need to be accomplished and switches after each task is accomplished. Tasks can be created, modified, or deleted as the pair progresses.Ideally each takes about the same amount of time to complete.
+
+## In this article, you learned to:
+*   Define pair programming and remote pair programming
+*   List at least five potential benefits of pair programming
+*   List at least five potential challenges of pair programming
+*   Describe the mechanics of the driver/navigator style of pair programming
+
+## Keep Learning
+1. Blog: [What’s the Best Way to Pair?](https://tanzu.vmware.com/content/blog/what-s-the-best-way-to-pair) by Maya Rosecrance and Sarah Connor.
+1. Blog: [5 Tips for Effective Remote Pair Programming While Working from Home](https://tanzu.vmware.com/content/blog/5-tips-for-effective-remote-pair-programming-while-working-from-home)
+1. Video: [I've Pair Programmed for 30,000 Hours: Ask Me Anything! — Joe Moore](https://www.youtube.com/watch?v=RCDfBioUgts)
+
```

Comment:
Maybe you could add a reference here to the great publicly available "Parrit" pair roation management tool our colleagues have built: https://parrit.io/

Reply:
done